### PR TITLE
Add test coverage for Node.js

### PR DESCRIPTION
**Pull request by sh1mmer**
cc @sectioneight @breerly @Raynos @kriskowal @Willyham 

I'd like to move to Tape but I wanted to get the coverage before I rewrote all the tests.
I'd also like to add testing server/client pairs from Python, Java, Go and any other languages we use somewhere maybe that doesn't belong in the Node branch though.

### CODE REVIEW

File: lib/nodejs/test/testAll.sh [hunk 1]
```diff
@@ -126,4 +175,10 @@ testWSClientServer binary framed || TESTOK=1
 testWSClientServer json buffered --promise || TESTOK=1
 testWSClientServer binary framed --ssl || TESTOK=1
 
+if [ -n "${COVER}" ]; then
+  ${DIR}/../node_modules/.bin/istanbul report --include "${DIR}/../coverage/report*/coverage.json" lcov cobertura
+  #rm -r ${DIR}/../coverage/report*/*
```
Comment (sectioneight) [inline lib/nodejs/test/testAll.sh]: remove commented out code?

**Comment by Raynos** [comment]
seems reasonable.

File: lib/nodejs/test/testAll.sh (hunk 1 above)
Comment (sh1mmer) [inline lib/nodejs/test/testAll.sh]: Oh, sorry I thought I'd added that back in. We probably do want to remove these once we've done the rollup report. I just disabled it while testing. Thanks for catching that.

**Comment by Raynos** [comment]
lgtm.

**Comment by breerly** [comment]
Lets move this to land against uber-master instead.

**Comment by sh1mmer** [comment]
Merged this into #7

**Pull request closed without merging**